diff --git a/node_modules/graphql/validation/rules/UniqueFieldDefinitionNamesRule.js b/node_modules/graphql/validation/rules/UniqueFieldDefinitionNamesRule.js
index f0d9fcb..fcb1291 100644
--- a/node_modules/graphql/validation/rules/UniqueFieldDefinitionNamesRule.js
+++ b/node_modules/graphql/validation/rules/UniqueFieldDefinitionNamesRule.js
@@ -48,14 +48,7 @@ function UniqueFieldDefinitionNamesRule(context) {
       const fieldName = fieldDef.name.value;
 
       if (hasField(existingTypeMap[typeName], fieldName)) {
-        context.reportError(
-          new _GraphQLError.GraphQLError(
-            `Field "${typeName}.${fieldName}" already exists in the schema. It cannot also be defined in this type extension.`,
-            {
-              nodes: fieldDef.name,
-            },
-          ),
-        );
+        // Ignore errors when fields are already defined
       } else if (fieldNames[fieldName]) {
         context.reportError(
           new _GraphQLError.GraphQLError(
