diff --git a/node_modules/@omnigraph/json-schema/cjs/addExecutionLogicToComposer.js b/node_modules/@omnigraph/json-schema/cjs/addExecutionLogicToComposer.js
index 9bbb265..491eb4c 100644
--- a/node_modules/@omnigraph/json-schema/cjs/addExecutionLogicToComposer.js
+++ b/node_modules/@omnigraph/json-schema/cjs/addExecutionLogicToComposer.js
@@ -49,6 +49,7 @@ async function addExecutionDirectivesToComposer(name, { schemaComposer, logger,
 >**Method**: \`${operationConfig.method}\`
 >**Base URL**: \`${endpoint}\`
 >**Path**: \`${operationConfig.path}\`
+>**Version**: \`${name.split('@')[1]}\`
 ${operationConfig.description || ''}
 `;
             }
diff --git a/node_modules/@omnigraph/json-schema/cjs/addRootFieldResolver.js b/node_modules/@omnigraph/json-schema/cjs/addRootFieldResolver.js
old mode 100644
new mode 100755
index 8006747..f2af17c
--- a/node_modules/@omnigraph/json-schema/cjs/addRootFieldResolver.js
+++ b/node_modules/@omnigraph/json-schema/cjs/addRootFieldResolver.js
@@ -35,6 +35,21 @@ function addHTTPRootFieldResolver(schema, field, logger, globalFetch, { path, op
         const interpolatedBaseUrl = string_interpolation_1.stringInterpolator.parse(endpoint, interpolationData);
         const interpolatedPath = string_interpolation_1.stringInterpolator.parse(path, interpolationData);
         let fullPath = (0, url_join_1.default)(interpolatedBaseUrl, interpolatedPath);
+        let subPath;
+
+        // HANDLE HATEOAS LINKS
+        if (root && root?.followLinks?.length) {
+          const index = info?.path?.key?.match(/_(\d+)_/)?.[1];
+          subPath = root.followLinks?.[index]?.followLink;
+          fullPath = interpolatedBaseUrl + subPath;
+        } else if (root && root.followLink) {
+          subPath = root.followLink;
+          fullPath = interpolatedBaseUrl + subPath;
+        }
+        if (root && !subPath) {
+          return;
+        }
+
         const headers = {};
         for (const headerName in globalOperationHeaders) {
             const nonInterpolatedValue = globalOperationHeaders[headerName];
