diff --git a/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js b/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js
index e815992..9084b8f 100644
--- a/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js
+++ b/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js
@@ -278,7 +278,7 @@ async function getJSONSchemaOptionsFromOpenAPIOptions(name, { source, fallbackFo
                     operationConfig.headers = operationConfig.headers || {};
                     operationConfig.headers.Accept = methodObj.produces.join(', ');
                 }
-                if ('content' in responseObj) {
+                if ('content' in responseObj && Object.keys(responseObj.content).length !== 0) {
                     const responseObjForStatusCode = {
                         oneOf: [],
                     };
