diff --git a/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js b/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js
index e815992..120f727 100644
--- a/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js
+++ b/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js
@@ -31,6 +31,16 @@ async function getJSONSchemaOptionsFromOpenAPIOptions(name, { source, fallbackFo
             readFileOrUrl: readFileOrUrlForJsonMachete,
             debugLogFn: logger.debug.bind(logger),
         }));
+        // Add a dummy property to each schema without properties
+        const schemas = oasOrSwagger.components?.schemas
+        if (schemas) {
+            Object.keys(schemas).forEach(schemaName => {
+                const schema = schemas[schemaName];
+                if (schema.type === 'object' && !schema.properties) {
+                    schema.properties = { "_null": {} };
+                }
+            });
+        }
     }
     else {
         oasOrSwagger = await (0, json_machete_1.dereferenceObject)(source, {
@@ -278,7 +288,7 @@ async function getJSONSchemaOptionsFromOpenAPIOptions(name, { source, fallbackFo
                     operationConfig.headers = operationConfig.headers || {};
                     operationConfig.headers.Accept = methodObj.produces.join(', ');
                 }
-                if ('content' in responseObj) {
+                if ('content' in responseObj && Object.keys(responseObj.content).length !== 0) {
                     const responseObjForStatusCode = {
                         oneOf: [],
                     };
