diff --git a/node_modules/graphql-jit/dist/execution.js b/node_modules/graphql-jit/dist/execution.js
old mode 100644
new mode 100755
index ee4f545..e0cbe09
--- a/node_modules/graphql-jit/dist/execution.js
+++ b/node_modules/graphql-jit/dist/execution.js
@@ -348,8 +348,9 @@ function compileType(context, parentType, type, fieldNodes, originPaths, destina
     let errorDestination;
     if ((0, graphql_1.isNonNullType)(type)) {
         type = type.ofType;
+        // FIXME: This is a hack for not handling error when type is notNull but the return value is null
         const nullErrorStr = `"Cannot return null for non-nullable field ${parentType.name}.${getFieldNodesName(fieldNodes)}."`;
-        body += `(${GLOBAL_NULL_ERRORS_NAME}.push(${createErrorObject(context, fieldNodes, previousPath, nullErrorStr)}), null) :`;
+        body += `(console.error(${nullErrorStr}), null) : `;
         errorDestination = GLOBAL_NULL_ERRORS_NAME;
     }
     else {
