diff --git a/node_modules/json-machete/cjs/healJSONSchema.js b/node_modules/json-machete/cjs/healJSONSchema.js
index 5d7e615..88ae5fb 100644
--- a/node_modules/json-machete/cjs/healJSONSchema.js
+++ b/node_modules/json-machete/cjs/healJSONSchema.js
@@ -40,12 +40,28 @@ exports.AnySchema = {
         { type: 'object', additionalProperties: true },
     ],
 };
+function removeUnnecessaryProperties(obj, seen = []) {
+    if (seen.includes(obj)) {
+        return;
+    }
+    seen.push(obj);
+    delete obj.pattern;
+    delete obj.maxLength;
+    delete obj.minLength;
+    for (let key in obj) {
+        if (typeof obj[key] === 'object' && obj[key] !== null) {
+            removeUnnecessaryProperties(obj[key], seen);
+        }
+    }
+}
 async function healJSONSchema(schema, debugLogFn) {
     const schemaByTitle = new Map();
     const anySchemaOneOfInspect = (0, cross_inspect_1.inspect)(exports.AnySchema.oneOf);
     return (0, visitJSONSchema_js_1.visitJSONSchema)(schema, {
         enter: async function healSubschema(subSchema, { path }) {
             if (typeof subSchema === 'object') {
+                // Remove the "minLength", "maxLength" and "pattern" properties of each scalar
+                removeUnnecessaryProperties(subSchema)
                 if (subSchema.title === 'Any' ||
                     (subSchema.oneOf && (0, cross_inspect_1.inspect)(subSchema.oneOf) === anySchemaOneOfInspect)) {
                     return exports.AnySchema;
@@ -360,8 +376,10 @@ async function healJSONSchema(schema, debugLogFn) {
                                 subSchema.maxLength ||
                                 subSchema.minLength ||
                                 subSchema.enum) {
-                                debugLogFn?.(`${path} has a pattern or maxLength or minLength or enum but no title. Setting it to ${pathBasedName}`);
-                                subSchema.title = pathBasedName;
+                                // Change the default enum names to be able to merge them
+                                const newEnumName = `ENUM_${maybeDefinitionBasedPath.split('/')[maybeDefinitionBasedPath.split('/').length-1]}`
+                                debugLogFn?.(`${path} has a enum but no title. Setting it to ${newEnumName}`);
+                                subSchema.title = newEnumName;
                                 // Otherwise use the format name
                             }
                             break;
